src/SFA.DAS.Tools.Servicebus.Support.Infrastructure/Services/ServiceBus/AsbService.cs

```diff
@@ -126,8 +126,20 @@ public async Task SendMessagesAsync(IEnumerable<QueueMessage> messages, string q
 
             if (messages.Count() > 0)
             {
-                var orginalMessages = messages.Select(m => m.OriginalMessage).ToList();
-                await messageSender.SendAsync(orginalMessages);
+                var orginalMessages = new List<Message>();
+                try
+                {
+                    orginalMessages = messages.Select(m => m.OriginalMessage).ToList();
+                    await messageSender.SendAsync(orginalMessages);
+                }catch(MessageSizeExceededException)
+                {
+                    _logger.LogDebug("SendMessagesAsync MessageSizeExceededException");
+                    foreach(var msg in orginalMessages)
+                    {
+                        await messageSender.SendAsync(msg);
+                    }                                                                
```

Comment:
This smells funny, doing any kind of processing inside a catch block could be problematic.
Is it worth implementing something like [Polly](http://www.thepollyproject.org/about/)?